### ISSUE TITLE
bug in `@diagnotic disable-line`

### ISSUE THREAD

**Issue by lewis6991** [CppCXY/emmylua-analyzer-rust#158]
<img width="677" alt="Image" src="https://github.com/user-attachments/assets/c57e4c35-11cf-468d-b112-d100ab5f084e" />

```lua
local a --- @deprecated

---@diagnostic disable-next-line: deprecated
local _b = a

local _c = a ---@diagnostic disable-next-line: deprecated

local _d = a ---@diagnostic disable-line: deprecated
```

`_c` should warn. `_d` should not.

**Comment by CppCXY** [comment]
Actually, I don't support disable-line, because I didn't see people using this when I was writing it.

**Comment by clason** [comment]
Also typo: "`depreacated`" :)
